npreadtext/_readers.py

```diff
@@ -210,8 +210,6 @@ def read(file, *, delimiter=',', comment='#', quote='"',
             raise ValueError(f"length of usecols ({len(usecols)}) and "
                              f"number of fields in dtype ({len(codes)}) "
                              "do not match.")
-        if len(dtypes) == 1 and usecols is not None:
-            dtypes *= len(usecols)
```

Comment:
I think I just had this wrong. The hole "homogeneous" logic probably could use some more simplification though.

However, it probably is *still* wrong (will open an issue).  Since `len(dtypes) != 1` is probably the wrong check.  What we really want to check for is "homogeneous", and that means that `len(dtypes) == 1 and dtypes[0] is dtype`.